indexer/src/main/java/org/uniprot/store/indexer/uniprotkb/converter/UniProtEntryConverter.java

```diff
@@ -133,6 +138,11 @@ public UniProtDocument convert(UniProtKBEntry source) {
         }
     }
 
+    private boolean hasIsoform(UniProtKBEntry source) {
```

Comment:
okay, if entry has alternative products comments meaning it has isoform. 
Does it means if entry has isoform, it will always has canonical isoform (and optional others)?

Reply:
**okay, if entry has alternative products comments meaning it has isoform.**
Yes
**Does it means if entry has isoform, it will always has canonical isoform (and optional others)?** 
Canonical is the entry itself (P12345 or P12345-1), isoforms are the others (P12345-2, P12345-3), when we have alternative products, it means that we will have canonical+isoforms.